File: vaccination-app/src/app/reservation.ts
import {VaccinationCenter} from "./vaccination-center";

export class Reservation {
  idReservation!: number;
  reservationDate: Date;
  center: VaccinationCenter;
  nom: string;
  prenom: string;
  email: string;
  approved: boolean;


  constructor(center: VaccinationCenter, nom:string, prenom: string, email: string, reservationDate: Date, approved: boolean) {
    this.center = center;
    this.nom = nom;
    this.prenom = prenom;
    this.email = email;
    this.reservationDate = reservationDate;
    this.approved = approved;
  }
}

File: covid-api/src/main/java/org/polytech/covid/Entities/Reservation.java
package org.polytech.covid.Entities;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.beans.factory.annotation.Value;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Date;

@Entity
@Data
@AllArgsConstructor @NoArgsConstructor
@Table(name = "reservation")
public class Reservation implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_reservation")
    private long idReservation;

    @Column(name = "reservation_date")
    private Date reservationDate;

    @ManyToOne @JoinColumn(name = "center")
    private VaccinationCenter center;

    private String nom;

    private String prenom;

    private String email;

    @Column(name = "validee")
    @Value("${false}")
    private boolean approved;


}
